src/js/components/FrameworkConfiguration.js

```diff
@@ -177,40 +179,44 @@ class FrameworkConfiguration extends Component {
   }
 
   getTermsAndConditions() {
-    const { packageDetails } = this.props;
+    const { packageDetails, i18n } = this.props;
 
     const termsUrl = packageDetails.isCertified()
       ? "https://mesosphere.com/catalog-terms-conditions/#certified-services"
       : "https://mesosphere.com/catalog-terms-conditions/#community-services";
 
-    return {
-      __html: `By running this service you agree to the <a href=${termsUrl} target='_blank' title='Terms and Conditions'>terms and conditions</a>.`
-    };
+    const termsAndConditions = i18n._(t`Terms and Conditions`);
+
+    return (
+      <Trans render="span">
+        By running this service you agree to the{" "}
+        <a href={termsUrl} target="_blank" title={termsAndConditions}>
+          terms and conditions
+        </a>.
+      </Trans>
+    );
   }
 
   getWarningMessage() {
-    const { packageDetails } = this.props;
+    const { packageDetails, i18n } = this.props;
 
     const preInstallNotes = packageDetails.getPreInstallNotes();
     if (preInstallNotes) {
-      const message = StringUtil.parseMarkdown(preInstallNotes);
-      message.__html = `<strong>Preinstall Notes: </strong>${message.__html} ${
-        this.getTermsAndConditions().__html
-      }`;
+      const notes = StringUtil.parseMarkdown(preInstallNotes);
 
       return (
-        <div
-          dangerouslySetInnerHTML={message}
-          className="pre-install-notes message message-warning"
-        />
+        <div className="pre-install-notes message message-warning">
+          <strong>{i18n._(t`Preinstall Notes`)}: </strong>
```

Comment:
Is there a reason we have to pre translate this strings instead of just using a `Trans` macro? (Same as line 241, 332)